### PR TITLE
Expose userAgent to the client

### PR DESCRIPTION
**Pull request by ramonjd**
This is an experimental branch to check the build size after adding `express-useragent` the client bundle.

I've replaced uses of `navigator.userAgent` with the lib.

Background: `p4TIVU-96y-p2`

If this branch makes sense we can revert the `navigator.userAgent` replaces I've made, and merge with just the thin client `client/lib/user-agent/index.js`

### 1. Importing into the client

a20d87b95416a5704b5aba65696492630a0855d0

```
Commit: a20d87b95416a5704b5aba65696492630a0855d0
Author: ramonjd
At: 2018-08-22T04:00:15Z
Message: replacing navigator.userAgent with lib
Delta:
chunk                                     stat_size           parsed_size           gzip_size
async-load-lib-rubberband-scroll-disable       -5 B  (-1.1%)        -36 B  (-8.0%)       -6 B  (-2.0%)
async-load-reader-sidebar                     +29 B  (+0.0%)        +20 B  (+0.1%)      -13 B  (-0.2%)
build                                        +606 B  (+0.0%)       +192 B  (+0.0%)      +79 B  (+0.0%)
comments                                      -18 B  (-0.0%)        -46 B  (-0.0%)      -31 B  (-0.1%)
manifest                                       +0 B                  +0 B                -2 B  (-0.0%)
settings-writing                              -48 B  (-0.0%)        -61 B  (-0.0%)      -40 B  (-0.1%)
themes                                        -13 B  (-0.0%)        -16 B  (-0.0%)       -9 B  (-0.0%)
vendors~build                              +31634 B  (+1.1%)     +16561 B  (+1.3%)    +4497 B  (+1.4%)
```

### 2. Adding to `window.app`

03e6fbb45f6bf8e0380270be6b74b8d20b66c90e

```
Commit: 03e6fbb45f6bf8e0380270be6b74b8d20b66c90e
Author: ramonjd
At: 2018-08-22T04:59:55Z
Message: adding userAgent to `window.app`
Delta:
chunk                                     stat_size           parsed_size           gzip_size
async-load-lib-rubberband-scroll-disable       -5 B  (-1.1%)        -36 B  (-8.0%)       -6 B  (-2.0%)
async-load-reader-sidebar                     +29 B  (+0.0%)        +20 B  (+0.1%)      -13 B  (-0.2%)
build                                        +524 B  (+0.0%)       +106 B  (+0.0%)      +53 B  (+0.0%)
comments                                      -18 B  (-0.0%)        -46 B  (-0.0%)      -31 B  (-0.1%)
manifest                                       +0 B                  +0 B                -2 B  (-0.0%)
settings-writing                              -48 B  (-0.0%)        -61 B  (-0.0%)      -40 B  (-0.1%)
themes                                        -13 B  (-0.0%)        -16 B  (-0.0%)       -9 B  (-0.0%)
```
http://iscalypsofastyet.com/branch?branch=try/add-user-agent-lib

## Results
Is an extra 4.5k (+1.4%) to vendors negligible? My instinct tells me that adding extra guff to the window object is bad, but we're not winning back many bytes in the modules.

### CODE REVIEW

File: server/pages/index.js [hunk 1]
```diff
@@ -245,13 +246,15 @@ function getDefaultContext( request ) {
 		store: createReduxStore( initialServerState ),
 		bodyClasses,
 		sectionCss,
+		userAgent: useragent.parse( request.headers[ 'user-agent' ] ),
```
Comment (lsl) [inline server/pages/index.js]: `request.headers[ 'user-agent' ]` - Is this always set / on all browsers?

**Review by lsl — COMMENTED** [review]
I'm biased, so I won't approve but LGTM - only question is how user-agent is being populated.

File: server/pages/index.js (hunk 1 above)
Comment (ramonjd) [reply server/pages/index.js]: Good question. It made me look this up.

Express [extends](https://github.com/expressjs/express/blob/b97faff6e2aa4d34d79485fe4331cb0eec13ad57/lib/request.js#L31) node's [HTTP module](https://nodejs.org/api/http.html#http_event_request). So a `headers` property will exist at least. 

But looking more closely, I realized I'd used `useragent.parse( request.headers[ 'user-agent' ] )` when I didn't need to at all.

We're adding [express-useragent](https://github.com/biggora/express-useragent) to the middleware in [boot.js](https://github.com/Automattic/wp-calypso/blob/master/server/boot/index.js#L30), which makes a parsed `req.useragent` object available on the request object.

Updated 👍

**Comment by ramonjd** [comment]
Closing this PR to act as a reference for #26931, which implements the clientside import.

**Pull request closed without merging**